collector/lib/CollectorConfig.cpp

```diff
@@ -422,33 +425,106 @@ void CollectorConfig::YamlConfigToConfig(YAML::Node& yamlConfig) {
   externalIpsConfig->set_enable(enableExternalIps);
 
   SetRuntimeConfig(runtime_config);
-  CLOG(INFO) << "Runtime configuration:";
-  CLOG(INFO) << GetRuntimeConfigStr();
+  CLOG(INFO) << "Runtime configuration:\n" + GetRuntimeConfigStr();
```

Comment:
C++ related question for me: Is there something like a `string` function which is called automatically when used in a formatter, eg in `prinftf("%s", runtimeConfig)`?

Reply:
In C++ we usually just use the output stream operator (`operator<<`) which handles formatted output, you usually overload it for your classes and then use it directly. There are stream manipulators for handling certain aspects of the output like whether `true` and `false` should be integers or alphanumeric (https://en.cppreference.com/w/cpp/io/manip).

C++23 added [std::format](https://en.cppreference.com/w/cpp/io/print) which is actually pretty nice, but because it is in C++23, it won't be used until 2032 in production projects.

Reply:
Before I forget, you can also just call `printf` from C++, as you can with any C function, it's just not best practices because it leads to writing "C style C++", which is terrible and you should definitely never do :pokerface:

Reply:
I am now just using `<<`.